misc/loop/run_loop.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -e # exit on error
+
+# Set up the kill signal callback
+teardown() {
+  echo "Stopping background processes..."
+  kill 0
+}
+
+# Helper for checking the local exit code
+check_exit_code() {
+  local exit_code=$?
+  if [ $exit_code -ne 0 ]; then
+    echo "Error: Process failed with exit code $exit_code"
+    teardown
```

Comment:
shouldn't teardown always be called, whatever is exit_code?

Reply:
You're completely right, it should always kill hanging processes:

[33b86d0](https://github.com/gnolang/gno/pull/1237/commits/33b86d0b6f9b0573bac32a19ff6f06369d58f08b)